.github/workflows/main.yml

```diff
@@ -224,3 +240,36 @@ jobs:
         env:
           DB: oracle
           COVERAGE_FILE: .coverage.oracle.${{ matrix.python-version }}
+      - uses: actions/upload-artifact@v3
+        with:
+          name: oracle-coverage-files
+          path: .coverage.*
+
+  generate-coverage-report:
+    runs-on: ubuntu-20.04
+    # ref: https://github.com/actions/runner/issues/491#issuecomment-850884422
+    if: |
+      always() &&
+      (needs.test-oracle.result == 'success' && needs.test-sqlite.result == 'success' && needs.test-mssql.result == 'success' && needs.test-mysql.result == 'success'  && needs.test-postgres.result == 'success')
```

Comment:
Instead of doing this, can we just send the coverage file to coverall in every job and let coverall do the merging ?